plugin/pkg/shoot/dnshostedzone/admission.go

```diff
@@ -197,6 +208,12 @@ func verifyHostedZoneID(shoot *garden.Shoot, secretBindingLister gardenlisters.S
 		if !accessKeyFound || !secretKeyFound {
 			return fmt.Errorf("specifying the `.spec.dns.hostedZoneID` field is only possible if the cloud provider secret or the secret referenced in .spec.dns.secretName contains credentials for AWS Route53 (%s and %s)", awsbotanist.AccessKeyID, awsbotanist.SecretAccessKey)
 		}
+	case garden.DNSAlicloud:
+		_, accessKeyFound := credentials.Data[alicloudbotanist.AccessKeyID]
+		_, accessSecretFound := credentials.Data[alicloudbotanist.AccessKeySecret]
+		if !accessKeyFound || !accessSecretFound {
+			return fmt.Errorf("specifying the `.spec.dns.hostedZoneID` field is only possible if the cloud provider secret or the secret referenced in .spec.dns.secretName contains credentials for Alicloud DNS (%s and %s)", alicloudbotanist.AccessKeyID, alicloudbotanist.AccessKeySecret)
```

Comment:
invalid error message? hosted zone id is not supported for alicloud

Reply:
I reused the the function verifyHostedZoneID() for Alicloud. It is quite misleading. I will do some refactory.